src/Core/Geometry/Image.cpp

```diff
@@ -61,6 +61,13 @@ Eigen::Vector2d Image::GetMaxBound() const
     return Eigen::Vector2d(width_, height_);
 }
 
+bool Image::TestImageBoundary(double u, double v,
+        double inner_margin = 0.0) const
+{
+    return (u >= inner_margin && u < width_ - inner_margin &&
+        v >= inner_margin && v < height_ - inner_margin);
+}
+
```

Comment:
I add this as it is frequently used in this PR. However, this function also can be in Geometry2D.h. Not quite sure this is the right place.